### PR TITLE
Update CI and dependencies

### PR DESCRIPTION
**Pull request by fasmat**
Part of https://github.com/spacemeshos/pm/issues/242

In preparation for the first release I made the following changes:

- add dependabot for go dependencies and github actions
- updated dependencies
- removed build on arm vitual image - is this still needed when we have arm servers now? @lrettig

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -1,63 +1,33 @@
...
-      run: go build -v ./...
-
-    - name: Test
-      run: go test -v ./...
-
-  build-on-arm-emu:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v3
-    - uses: pguyot/arm-runner-action@v2.5.2
```
[the remark this answers is not in the document]
Comment (fasmat) [reply .github/workflows/go.yml:38]: This action uses the `raspios_lite_arm64` image, which emulates `arm64`/`aarch64`, the same architecture as `[self-hosted, linux-arm64]`. Do you expect go to behave differently on Raspberry Pi OS compared to Ubuntu (both are based on Debian)?

[the remark this answers is not in the document]
Comment (lrettig) [reply .github/workflows/go.yml:38]: Maybe the right thing to do here is to compile and run on an emulated 32 bit rpi CPU. We could use `base_image: raspios_lite:latest` (or just remove this entirely as it's the default); see https://github.com/pguyot/arm-runner-action#base_image. If we want to be _really_ thorough we could use a matrix like the one [here](https://github.com/pguyot/arm-runner-action#32-and-64-bits) but I think a single 32 bit system is enough. Point is, there are several ARM architectures! And they [handle floats differently](https://stackoverflow.com/questions/37790029/what-is-difference-between-arm64-and-armhf).

**Comment by fasmat** [comment]
On 32 bit systems quite a lot of tests are failing - how to proceed @lrettig ?

**Comment by fasmat** [comment]
I created an issue for the failing 32 bit tests here: https://github.com/spacemeshos/economics/issues/13 (references issue in `github.com/ericlagergren/decimal` and updated the README.md with information about supported platforms

**Review by lrettig — APPROVED** [review]

**Pull request merged**